### PR TITLE
Merge develop into master

### LINKED ISSUE #16: Add social links to the footer
Add link to the ColoradoSchoolOfMines org as well as (optionally) social media sites (Facebook, etc.).

A few design thoughts:
- Maybe these links should be a configurable option from the admin page?
- These links should be just the logo as Fontawesome fonts.

### PR DESCRIPTION
**Pull request by sumnerevans**
@jackrosenthal , included in this PR:
- #37 (Resolves #16) - My changes to add social link ability to the footer
- #49 - Liam's changes to the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ This is the website for the Colorado School of Mines' local chapter for the
 It is written in Python and uses the powerful web framework
 [TurboGears](http://turbogears.com/).
 
+The steps below assume that you know how to use the [Linux command
+line](https://github.com/ColoradoSchoolOfMines/ACM-Guide-list/blob/master/computer_science/Command-Line.md)
+and how use git and GitHub.
+
```
Comment (jackrosenthal) [inline README.md]: I don't really think this is necessary for a README.

File: README.md [hunk 2]
```diff
@@ -28,11 +32,14 @@ Setup the application:
 
     $ cp development.ini.sample development.ini
     $ gearbox setup-app
+If `gearbox setup-app` fails, try specifying `pip3` in the previous two steps.
```
Comment (jackrosenthal) [inline README.md]: This is really system dependent... on some systems you'll even need something like ``python3.6 -m pip``. It's not really the job of our README to tell people how to use their system, and therefore this should not be included.

File: README.md [hunk 3]
```diff
@@ -28,11 +32,14 @@ Setup the application:
...
 
-Finally, serve the application:
+Serve the application:
 
     $ gearbox serve --reload --debug
 
+Finaly, go to your web browser and navigate to `localhost:8080` to view the website.
+
 ### To use a virtual environment
```
Comment (jackrosenthal) [inline README.md]: How did virtual env stuff end up in the README? If people choose to use this software, they should reference the official documentation on how to use it. Please remove; at most include a link to the documentation as a suggestion of an alternative development environment.

File: acmwebsite/templates/master.xhtml [hunk 4]
```diff
@@ -12,7 +12,7 @@
   </head>
 
   <body>
-    <!-- Navbar -->
+    <!--! Navbar -->
```
Comment (jackrosenthal) [inline acmwebsite/templates/master.xhtml]: These comments were here to let you easily navigate the HTML output of the page (were not necessarily just comments for the XML markup)... please revert.

File: acmwebsite/templates/master.xhtml [hunk 5]
```diff
@@ -61,7 +61,7 @@
     </div>
     <div class="container site-content">
 
-      <!-- Flash messages -->
+      <!--! Flash messages -->
```
Comment (jackrosenthal) [inline acmwebsite/templates/master.xhtml]: revert

File: acmwebsite/templates/master.xhtml [hunk 6]
```diff
@@ -70,18 +70,29 @@
         </div>
       </py:with>
 
-      <!-- Main included content -->
+      <!--! Main included content -->
```
Comment (jackrosenthal) [inline acmwebsite/templates/master.xhtml]: revert

**Comment by jackrosenthal** [comment]
As an added note, the README should be converted to reStructuredText before any further changes are made to it. (see the ``CONTRIBUTING.rst`` file, this is a requirement for all documentation related to this project)

File: README.md (hunk 3 above)
Comment (sumnerevans) [reply README.md]: This was added ages ago.

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+==========
+
+This website is a website for clubs and student organizations.
+
+Mozzarella It is written in Python and uses the powerful web framework
+`TurboGears`.
+
+.. TurboGears: http://turbogears.com/
+
+Getting Started
```
Comment (jackrosenthal) [inline README.rst]: Confusing header... is this where I start to set it up in production?

"Setting up your Development Environment" is better.

File: README.rst [hunk 8]
```diff
@@ -0,0 +1,43 @@
+Mozzarella
+==========
+
+This website is a website for clubs and student organizations.
+
+Mozzarella It is written in Python and uses the powerful web framework
+`TurboGears`.
+
+.. TurboGears: http://turbogears.com/
```
Comment (jackrosenthal) [inline README.rst]: See the reStructuredText specification on how to create links::

    powerful web framework TurboGears_.

    .. _TurboGears: http://turbogears.org

http://docutils.sourceforge.net/docs/user/rst/quickref.html

File: README.rst [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+Getting Started
+---------------
+
+These steps are meant to give you an overview of how to get a development
+environment set up for developing this application. Please adapt to your system
+as necessary. `This guide` gives more in-depth instructions on how to start
+working on a TurboGears project.
+
+.. This guide: https://github.com/ColoradoSchoolOfMines/ACM-Guide-list/blob/master/computer_science/TurboGears.md
```
Comment (jackrosenthal) [inline README.rst]: Again, this is not valid syntax.

Comment (jackrosenthal) [reply README.rst]: Also, why not link to the official TG documentation? If you have improvements to make to the TG documentation, you can submit a PR there (they are good about accepting them)...

File: development.ini.sample [hunk 10]
```diff
@@ -60,6 +60,11 @@ mailman.url = https://mailman.mines.edu/mailman/admin/acmx
 mailman.pipermail.url = https://mailman.mines.edu/pipermail/acmx
 mailman.secret = getthisfromjack
 
+# Social Links
+social_links = github,facebook
```
Comment (jackrosenthal) [inline development.ini.sample]: Why not just see which keys are defined? No need for this config option...

File: README.rst (hunk 8 above)
Comment (sumnerevans) [reply README.rst]: Site is down:

"Project web is currently offline pending the final migration of its data to our new datacenter."

File: development.ini.sample (hunk 10 above)
Comment (sumnerevans) [reply development.ini.sample]: I asked for feedback on this ages ago: https://github.com/ColoradoSchoolOfMines/mozzarella/pull/37#issue-159731128.

I couldn't find any way to check which keys are defined. Any suggestions would be appreciated.

File: README.rst (hunk 9 above)
Comment (sumnerevans) [reply README.rst]: @jackrosenthal , the point of this is for onboarding people onto the project. That is our number one problem: getting people up to speed quickly. The TG documenation does not do a good job of that. It requires some prior knowledge of what web development is. Our goal with these guides is to help people who have never worked on any web development project get up and running on TurboGears projects. This guide also links directly to the documentation fot TG for after they are set up and understand a little bit of what they are doing.

File: development.ini.sample (hunk 10 above)
Comment (jackrosenthal) [reply development.ini.sample]: > I couldn't find any way to check which keys are defined. Any suggestions would be appreciated.

``tg.config`` is like a dictionary (in fact, it inherits from ``collections.abc.MutableMapping``), so you can just call ``.keys()`` on it.

File: README.rst [hunk 11]
```diff
@@ -4,19 +4,19 @@ Mozzarella
...
+.. _TurboGears: http://turbogears.com/
 
-Getting Started
----------------
+Setting up your Development Environment
+---------------------------------------
 
 These steps are meant to give you an overview of how to get a development
 environment set up for developing this application. Please adapt to your system
 as necessary. `This guide` gives more in-depth instructions on how to start
```
Comment (jackrosenthal) [inline README.rst]: Do not assume that references are the default role.

File: development.ini.sample (hunk 10 above)
Comment (jackrosenthal) [reply development.ini.sample]: Want me to fix this or are you working on it?

File: README.rst (hunk 9 above)
Comment (jackrosenthal) [reply README.rst]: The guide you linked seemed to give a near-exact replica of what is in the README... just remove this entirely then, or link to the TurboGears docs (which gives a pretty good explanation of the virtualenv stuff): http://turbogears.readthedocs.io/en/latest/turbogears/minimal/index.html

File: development.ini.sample (hunk 10 above)
Comment (sumnerevans) [reply development.ini.sample]: I'm on it. Commit incomming...

**Comment by jackrosenthal** [comment]
OK if I make some grammar/style changes?

Also, let's target Python 3.4+, unless we have a good reason for new features from 3.5.

**Comment by jackrosenthal** [comment]
Okay, review my changes and then we can merge!

**Comment by sumnerevans** [comment]
Looks good! Merging.

**Pull request merged**